### PR TITLE
Use `ruff format` to format python files for faster formatting

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/11046?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11046/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11046
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Use ruff format to format python files for faster formatting:

image

source: https://astral.sh/blog/the-ruff-formatter

When formatting a file with > 1000 lines:

ruff: Too fast, no delay at all.
black: there is a small delay until formatting completes.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/autoformat.yml [hunk 1]
```diff
@@ -115,8 +115,8 @@ jobs:
       # ************************************************************************
       - if: steps.diff.outputs.python == 'true'
         run: |
-          ruff --fix .
-          black .
+          ruff check --fix .
+          ruff format .
           blacken-docs $(git ls-files '*.py' '*.rst' '*.md') || true
```
Comment (harupy) [inline .github/workflows/autoformat.yml:120]: We can remove blacken-docs once https://github.com/astral-sh/ruff/discussions/9871 is resolved.

File: conftest.py [hunk 2]
```diff
@@ -227,9 +227,7 @@ def pytest_collection_modifyitems(session, config, items):  # pylint: disable=un
 
 
 @pytest.hookimpl(hookwrapper=True)
-def pytest_terminal_summary(
-    terminalreporter, exitstatus, config
-):  # pylint: disable=unused-argument
+def pytest_terminal_summary(terminalreporter, exitstatus, config):  # pylint: disable=unused-argument
```
Comment (harupy) [inline conftest.py:230]: ruff ignores `# pylint: disable=unused-argument` when calculating the line length.

File: .github/workflows/autoformat.yml [hunk 3]
```diff
@@ -115,8 +115,8 @@ jobs:
       # ************************************************************************
       - if: steps.diff.outputs.python == 'true'
         run: |
-          ruff --fix .
-          black .
+          ruff check --fix .
+          ruff format .
```
Comment (serena-ruan) [inline .github/workflows/autoformat.yml:119]: Should we disable those lint rules when using format https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules

Comment (harupy) [reply .github/workflows/autoformat.yml:119]: I don't think we enable these conflicting rules

Comment (harupy) [reply .github/workflows/autoformat.yml:119]: If we did, ruff would warn us but I don't see any warnings.

**Review by serena-ruan — APPROVED** [review]
LGTM!

**Pull request merged**